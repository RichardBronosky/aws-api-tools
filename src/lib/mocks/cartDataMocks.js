export const mockCartItem1 = {
  activeAddons: null,
  price: 279.99,
  product: {
    title: 'Adaire Brown Side Table',
    sku: '23122360',
    price: 279.99,
    list_price: 349.99,
    delivery_type: 'O',
    category: 'tables',
    free_shipping: true,
    on_sale: true,
    primary_image:
      'http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090',
    shipping_cost_code: { FL: 'VD71', SE: 'VD71', TX: 'VD71' },
    strikethrough: { FL: true, OM: true, SE: true, TX: true },
  },
  sku: '23122360',
  quantity: 1,
}

export const mockCartItem2 = {
  sku: '1141677P',
  product: {
    title: 'Laney Park Dark Gray 7 Pc Sectional',
    sku: '1141677P',
    price: 1199.99,
    list_price: 1299.99,
    delivery_type: 'D',
    category: 'livingroom',
    free_shipping: false,
    on_sale: true,
    primary_image:
      'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
    package_skus: {
      SE: [
        { sku: '12216760', quantity: 1, price: '208.43' },
        { sku: '12116768', quantity: 4, price: '172.94' },
        { sku: '12316762', quantity: 1, price: '231.63' },
        { sku: '12516766', quantity: 1, price: '68.17' },
      ],
      TX: [
        { sku: '12316762', quantity: '1', price: '231.63' },
        { sku: '12216760', quantity: '1', price: '208.43' },
        { sku: '12516766', quantity: '1', price: '68.17' },
        { sku: '12116768', quantity: '4', price: '172.94' },
      ],
      FL: [
        { sku: '12316762', quantity: '1', price: '248.65' },
        { sku: '12116768', quantity: '4', price: '185.62' },
        { sku: '12216760', quantity: '1', price: '223.71' },
        { sku: '12516766', quantity: '1', price: '73.16' },
      ],
    },
    shipping_cost_code: { FL: 'ROOM', SE: 'ROOM', TX: 'ROOM' },
  },
  price: 1199.99,
  activeAddons: null,
  quantity: 3,
}

export const mockCart = {
  cartItems: [mockCartItem1, mockCartItem2],
}
