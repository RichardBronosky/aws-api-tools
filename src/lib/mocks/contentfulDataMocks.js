export const Banner1 = {
  contentfulContentGroup: {
    groupContent: [
      {
        __typename: 'ContentfulBanner',
        title: 'Promo E : Umbrella Sale 1 - C6 R1',
        startDate: null,
        endDate: null,
        displayContent: true,
        contentText: {
          childMarkdownRemark: {
            html:
              '<br>\n<span style="color:#FFFFFF; font-weight: bold; font-size: 28px;">GET A FREE UMBRELLA</span>\n<br>\n<span style="color:#FFFFFF; font-weight: bold; font-size: 28px;">WITH SELECT* OUTDOOR PURCHASES OF $999+</span>\n<br>\n<br>\n<span style="color:#FFFFFF; font-weight: bold;">*EXCLUDES VENDOR DIRECT.</span>\n<br>',
          },
        },
        contentHorizontalRuleColor: null,
        textPosition: 'center',
        contentPosition: 'center',
        containerSize: '50%',
        contentLinkColor: 'white',
        link: {
          id: '12703942-b7d5-5bc8-b79b-ad18d5321106',
          text: 'Link : Outdoor Patio',
          displayText: 'Outdoor',
          textColor: null,
          internalUrl: null,
          slug: '/furniture/outdoor-patio',
          url: null,
          altDesc: 'View All Outdoor Patio Furniture',
          title: null,
          type: null,
          target: null,
        },
        contentLink: null,
        buttons: null,
        imageAltText: 'Shop Umbrella Sale',
        desktopBackgroundImage: null,
        backgroundColor: 'Blue',
        mobileImage: {
          id: '4e8177a3-6c27-5544-a5ee-fcb384528020',
          sizes: {
            srcSet:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png?w=640&h=600&q=50 640w',
            src:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png?q=50',
            sizes: '(max-width: undefinedpx) 100vw, undefinedpx',
          },
          file: {
            url:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png',
            fileName: 'AnnSale_Umbrella1_MOBILE_640x600.png',
            contentType: 'image/png',
          },
          resolutions: { width: null, height: null },
        },
        desktopImage: {
          id: '8740a3af-1f18-5f98-9335-137b1f18a7c7',
          sizes: {
            srcSet:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png?w=1660&h=350&q=50 1660w',
            src:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png?q=50',
            sizes: '(max-width: undefinedpx) 100vw, undefinedpx',
          },
          file: {
            url:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png',
            fileName: 'AnnSale_Umbrella1_CONTAINER_1660x350.png',
            contentType: 'image/png',
          },
          resolutions: { width: null, height: null },
        },
        contentPadding: null,
        contentful_id: '3fezOgXNiGTW87ngJIx9cB',
      },
    ],
  },
}
