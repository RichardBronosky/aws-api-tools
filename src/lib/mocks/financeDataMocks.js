export const mockFinancePlans1 = [
  {
    financeCode: '50EQUALPMTS',
    effectiveDate: '2019-06-03 00:00:00',
    creditEndsDate: '2023-08-01 00:00:00',
    downPaymentRequired: true,
    termsAndConditions: {
      childMarkdownRemark: {
        html:
          'Finance Plan:<br><br>Promotion Type: <strong>No Interest until August 2023*</strong><br>Promotion Duration: <strong>Until August 2023*</strong><br>Promotional APR: 0%<br>Purchase APR: 29.99% <br>Down payment of sales tax and delivery required<br><br><br><br><strong>No Interest/Equal Payments</strong><br>On promotional purchase, monthly payments equal to the promotional purchase divided equally by the number of months in the promotional period are required until expiration or termination of promotion, but no Finance Charges will be assessed during Promotion Duration indicated if all minimum monthly payments on account paid when due.<br>The Purchase APR as indicated applies to expired promotions and optional charges. If minimum monthly payments are 60 days past due, the promotion may be terminated and a Penalty APR may apply. Promotional purchases on existing accounts may not receive full benefit of promotion terms, including reduced APR if applicable, if account is subject to penalty pricing. Payments over the minimum will be applied as required by applicable law. If you receive a change in terms of penalty rate notice that becomes effective before the delivery/installation date of your purchase, the new Purchase APR or Penalty APR (as applicable) will apply to the purchase balance added to your account upon delivery/installation. Subject to approval by Synchrony Bank.<br>',
      },
    },
    priority: '1',
    promoMessage: {
      childMarkdownRemark: {
        html:
          '<p><b>No Interest until August 2023*</b></p><p>On purchases priced at $499.99 and up made with your Rooms To Go credit card through 6/17/19. Equal Monthly Payments required until August 2023*.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p><p>*Promotional Offer applies only to single-receipt qualifying purchases of $499.99 and up. No interest until June 2021 on single-receipt qualifying purchases less than $499.99. No interest will be charged on promo purchase (including related optional credit insurance/debt cancellation charges) and equal monthly payments are required equal to initial promo purchase amount divided equally by the number of months in promo period until promo is paid in full. The equal monthly payment will be rounded to the next highest whole dollar and may be higher than the minimum payment that would be required if the purchase was a non-promotional purchase. Any monthly payments shown are in connection with the 60 months promotional offer, exclude taxes and delivery, and should allow you to pay off the promotional purchase within the promo period if (1) you make your payments by the due date each month and (2) this is the only balance on your account during the promo period.  If you have other balances on your account, this monthly payment will be added to the minimum payment applicable to those balances.  Regular account terms apply to non-promotional purchases. For new accounts: Purchase APR is 29.99. Minimum Interest Charge is $2. Existing cardholders should see their credit card agreement for their applicable terms. Subject to credit approval.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p>',
      },
    },
    threshold: '499.99',
    numberOfMonths: '50',
    siteFriendlyLabel: 'No Interest until August 2023*',
  },
  {
    financeCode: '90015-0621',
    effectiveDate: '2019-06-03 00:00:00',
    creditEndsDate: '2021-06-01 00:00:00',
    downPaymentRequired: true,
    termsAndConditions: {
      childMarkdownRemark: {
        html:
          'Finance Plan:<br><br>Promotion Type: <strong>No Interest until June 2021*</strong><br>Promotion Duration: <strong>Until June 2021*</strong><br>Promotional APR: 0%<br>Purchase APR: 29.99% <br>Down payment of sales tax and delivery required<br><br><br><br><strong>No Interest/Equal Payments</strong><br>On promotional purchase, monthly payments equal to the promotional purchase divided equally by the number of months in the promotional period are required until expiration or termination of promotion, but no Finance Charges will be assessed during Promotion Duration indicated if all minimum monthly payments on account paid when due.<br>The Purchase APR as indicated applies to expired promotions and optional charges. If minimum monthly payments are 60 days past due, the promotion may be terminated and a Penalty APR may apply. Promotional purchases on existing accounts may not receive full benefit of promotion terms, including reduced APR if applicable, if account is subject to penalty pricing. Payments over the minimum will be applied as required by applicable law. If you receive a change in terms of penalty rate notice that becomes effective before the delivery/installation date of your purchase, the new Purchase APR or Penalty APR (as applicable) will apply to the purchase balance added to your account upon delivery/installation. Subject to approval by Synchrony Bank.<br>',
      },
    },
    priority: '0',
    promoMessage: {
      childMarkdownRemark: {
        html:
          '<p><b>No Interest until August 2023*</b></p><p>On purchases priced at $499.99 and up made with your Rooms To Go credit card through 6/17/19. Equal Monthly Payments required until August 2023*.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p><p>*Promotional Offer applies only to single-receipt qualifying purchases of $499.99 and up. No interest until June 2021 on single-receipt qualifying purchases less than $499.99. No interest will be charged on promo purchase (including related optional credit insurance/debt cancellation charges) and equal monthly payments are required equal to initial promo purchase amount divided equally by the number of months in promo period until promo is paid in full. The equal monthly payment will be rounded to the next highest whole dollar and may be higher than the minimum payment that would be required if the purchase was a non-promotional purchase. Any monthly payments shown are in connection with the 60 months promotional offer, exclude taxes and delivery, and should allow you to pay off the promotional purchase within the promo period if (1) you make your payments by the due date each month and (2) this is the only balance on your account during the promo period.  If you have other balances on your account, this monthly payment will be added to the minimum payment applicable to those balances.  Regular account terms apply to non-promotional purchases. For new accounts: Purchase APR is 29.99. Minimum Interest Charge is $2. Existing cardholders should see their credit card agreement for their applicable terms. Subject to credit approval.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p>',
      },
    },
    marketingMessage: {
      markdown: {
        childMarkdownRemark: {
          html: 'No Interest until August 2023* marketing message',
        },
      },
    },
    threshold: '.00',
    numberOfMonths: '0',
    siteFriendlyLabel: 'No Interest until June 2021*',
  },
  {
    financeCode: '90015-0621',
    effectiveDate: '2019-06-03 00:00:00',
    creditEndsDate: '2021-06-01 00:00:00',
    downPaymentRequired: false,
    termsAndConditions: {
      childMarkdownRemark: {
        html:
          'Finance Plan:<br><br>Promotion Type: <strong>>No Interest until June 2021*</strong><br>Promotion Duration: <strong>Until  June 2021*</strong><br>Promotional APR: 0%<br>Purchase APR: 29.99% <br><br>Rooms To Go requires No Down Payment.<br><br><strong>No Interest/Equal Payments</strong><br>On promotional purchase, monthly payments equal to the promotional purchase divided equally by the number of months in the promotional period are required until expiration or termination of promotion, but no Finance Charges will be assessed during Promotion Duration indicated if all minimum monthly payments on account paid when due.<br>The Purchase APR as indicated applies to expired promotions and optional charges. If minimum monthly payments are 60 days past due, the promotion June be terminated and a Penalty APR June apply. Promotional purchases on existing accounts June not receive full benefit of promotion terms, including reduced APR if applicable, if account is subject to penalty pricing. Payments over the minimum will be applied as required by applicable law. If you receive a change in terms of penalty rate notice that becomes effective before the delivery/installation date of your purchase, the new Purchase APR or Penalty APR (as applicable) will apply to the purchase balance added to your account upon delivery/installation. Subject to approval by Synchrony Bank.<br>',
      },
    },
    priority: '2',
    promoMessage: {
      childMarkdownRemark: {
        html:
          '<p><b>24 MONTHS Interest Free Financing*</b></p><p>On purchases made with your Rooms To Go credit card through 6/17/19. Equal Monthly Payments Required until June 2021.*</p><p>Rooms To Go requires No Down Payment.</p><p>*Offer applies only to single-receipt qualifying purchases made on your Rooms To Go credit card made through 6/17/19. No interest will be charged on promo purchase and equal monthly payments are required equal to initial promo purchase amount divided equally by the number of months in promo period until promo is paid in full. The equal monthly payment will be rounded to the next highest whole dollar and may be higher than the minimum payment that would be required if the purchase was a non-promotional purchase. Regular account terms apply to non-promotional purchases. For new accounts: Purchase APR is 29.99%; Minimum Interest Charge is $2. Existing cardholders should see their credit card agreement for their applicable terms. Subject to credit approval.</p><p>Rooms To Go requires No Down Payment.</p>',
      },
    },
    threshold: '.00',
    numberOfMonths: '0',
    siteFriendlyLabel: '24 Months Interest Free Financing*',
  },
]
