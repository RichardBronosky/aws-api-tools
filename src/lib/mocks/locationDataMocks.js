export const mockLocation1 = {
  lat: 27.9836,
  long: -82.2756,
  city: 'Seffner',
  state: 'Florida',
  county: 'Hillsborough',
  zip: '33584',
  country_code: 'US',
  region: 'FL',
  delivery_zone: 'A',
  price_zone: '0',
  distribution_index: '1',
}

export const mockLocation2 = {
  lat: 33.92904,
  long: -84.3467,
  city: 'Dunwoody',
  state: 'Georgia',
  county: 'Fulton',
  zip: '30346',
  country_code: 'US',
  region: 'SE',
  delivery_zone: 'A',
  price_zone: '2',
  distribution_index: '10',
}
