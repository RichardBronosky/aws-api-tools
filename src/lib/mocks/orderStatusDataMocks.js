export const orderStatus = {
  AbleToReschedule: false,
  AdditionalDirections: '**** ALL DAY DELIVERY**** ****NO TIME FRAMES****',
  AffiliateData: '',
  Balance: '0.00',
  BillingAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    State: 'GA',
    Zip: '30328',
  },
  Customer: {
    AllowText2AltPhoneNumber: false,
    AllowText2PhoneNumber: false,
    AltPhoneNumber: '',
    CMKey: '01*202/292-8491',
    CustomerID: 'SE006284958',
    Division: 'SE',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    PhoneNumber: '202/292-8491',
  },
  DeliveryAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    Plus4: '',
    State: 'GA',
    Zip: '30328',
  },
  DeliveryCharge: '119.99',
  DeliveryDate: '2019-08-06T00:00:00.000Z',
  DeliveryType: 'E',
  Deposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '37.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'VA',
      },
    ],
  },
  Division: 'SE',
  EmailAddress: 'jamiamorrow@gmail.com',
  EstimatedDeliveryTimeFrame: '',
  FinDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '600.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'RGE',
      },
    ],
  },
  FinanceAccountNumber: '6019191263030424',
  FinancePlan: '90015-0121',
  FraudFlag: '',
  GCDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '1.95',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'GC',
      },
    ],
  },
  I95ind: false,
  LineItems: {
    LineItem: [
      {
        ChildItems: {
          LineItem: [
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36111639',
              UnitPrice: '193.24',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36211631',
              UnitPrice: '250.76',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
          ],
        },
        PackagePrice: '444',
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '3611163P',
        UnitPrice: '0',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Louie Gray Twin Loft Bed with Desk and Bookcase',
      },
      {
        ChildItems: null,
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '38096613',
        UnitPrice: '29',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Mattie Gray Chevron Bean Bag Chair',
      },
    ],
  },
  OrderDate: '2019-08-04T23:17:03.000Z',
  OrderMessage: 'OK',
  OrderNumber: '14000136I',
  OrderTotal: '638.95',
  PickUpData: {
    Phone: '',
    PickupAddress: { Address1: '', Address2: '', City: '', State: '', Zip: '' },
    PickupHours: { string: null },
    StoreName: '',
    StoreNumber: '980',
  },
  RemoteDeliveryDate: '0001-01-01T00:00:00.000Z',
  RemoteEstimatedDeliveryTimeFrame: '',
  ShipDate: '',
  Status: 'Open',
  StoreNumber: '980',
  TaxAmount: '45.96',
  TaxAreaId: '111211811',
  Active: true,
  DispatchTrack: 'https://fgo-rtg.dispatchtrack.com/track_order/f0d44a2/SE14000136I',
}

export const orderStatusList = [orderStatus]
