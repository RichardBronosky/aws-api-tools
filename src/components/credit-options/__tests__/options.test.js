import React from 'react'
import ReactModal from 'react-modal'
import renderer from 'react-test-renderer'
import CreditOptions from '../options'
import { mockFinancePlans1 } from '../../../lib/mocks/financeDataMocks'

ReactModal.setAppElement('*')

describe('CreditOptions', () => {
  const testProps = {
    promoModalOpen: false,
    postbackRefId: 'test-id',
    accountError: false,
    financePlan: mockFinancePlans1[0],
    endDate: new Date('2023-08-04'),
    onKeyDown: jest.fn(),
    setModalOpen: jest.fn(),
    accountSubmit: jest.fn(),
    setAccount: jest.fn(),
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<CreditOptions { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with an account error', () => {
    const tree = renderer.create(<CreditOptions { ...{ ...testProps, accountError: 'test error' } } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when finance plan has 0 as threshold', () => {
    const tree = renderer
      .create(<CreditOptions { ...{ ...testProps, financePlan: { ...mockFinancePlans1[0], threshold: '.00' } } } />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls setModalOpen with true when simulating more info banner click', () => {
    const tree = renderer.create(<CreditOptions { ...testProps } />)
    tree.root.findByProps({ className: 'link no-margin' }).props.onClick()
    expect(testProps.setModalOpen).toHaveBeenCalledWith(true)
  })

  it('calls setModalOpen with false when simulating close banner', () => {
    const tree = renderer.create(<CreditOptions { ...{ ...testProps, promoModalOpen: true } } />)
    tree.root.findByProps({ promoMessage: testProps.financePlan.promoMessage }).props.closeModal()
    expect(testProps.setModalOpen).toHaveBeenCalledWith(false)
  })

  it("calls onKeyDown with 'test' when simulating key down", () => {
    const tree = renderer.create(<CreditOptions { ...testProps } />)
    tree.root.findByProps({ className: 'login-field' }).props.onKeyDown('test')
    expect(testProps.onKeyDown).toHaveBeenCalledWith('test')
  })

  it("calls setAccount with '1234' when simulating account number input", () => {
    const tree = renderer.create(<CreditOptions { ...testProps } />)
    tree.root.findByProps({ className: 'login-field' }).props.onChange({ target: { value: '1234' } })
    expect(testProps.setAccount).toHaveBeenCalledWith('1234')
  })

  it('calls accountSubmit when submit button is clicked', () => {
    const tree = renderer.create(<CreditOptions { ...testProps } />)
    tree.root.findByProps({ className: 'account-btn' }).props.onClick()
    expect(testProps.accountSubmit).toHaveBeenCalled()
  })
})
