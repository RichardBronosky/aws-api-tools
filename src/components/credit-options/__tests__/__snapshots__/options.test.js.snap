// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditOptions renders correctly when finance plan has 0 as threshold 1`] = `
<div
  className="grid-container"
>
  <div
    className="grid-x card options-info"
  >
    <div
      className="small-12 cell grid-x"
    >
      <div
        className="cell medium-4"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="undefinedshopOnline.jpg"
        />
      </div>
      <div
        className="cell medium-8"
      >
        <div
          className="top-info"
        >
          <h2>
            Open A Rooms To Go Account - Credit Extended by Synchrony Bank
          </h2>
          <p>
            With a Rooms To Go account, you will be able to take advantage of exclusive financing offers in your area. Click below to fillout our online application.
          </p>
          <div
            className="cell medium-6 "
          >
            <form
              id="creditApplnForm"
              method="POST"
              name="frmApplicationTermsCond"
              target="_blank"
            >
              <input
                id="uniqueId"
                name="uniqueId"
                type="hidden"
              />
              <input
                id="postbackRefId"
                name="postbackRefId"
                type="hidden"
                value="test-id"
              />
              <input
                name="pcgc"
                type="hidden"
              />
              <input
                name="mid"
                type="hidden"
              />
              <input
                aria-label="Continue with online application (opens in new window)"
                className="blue-action-btn"
                data-seo="applyFinance"
                id="applyFinance"
                role="link"
                type="submit"
                value="Continue with online application"
              />
            </form>
            <p>
              Please note that online finance offers may be different from advertised credit promotions available in showrooms.
            </p>
          </div>
        </div>
        <hr />
        <div
          className="mid-banner"
        >
          <button
            className="link no-margin"
            onClick={[Function]}
          >
            <span
              className="plantitle"
            >
              Interest free financing for 60 months*
            </span>
            <span
              className="plantext"
            >
              On purchases priced at $999.99 and up made with your Rooms To Go credit card through 10/14/19. 
            </span>
            <span
              className="plantext"
            >
              Equal Monthly Payments required for 60 months.* Monthly payments shown are only applicable with this special financing offer.
            </span>
            <span
              className="plantext"
            >
              Rooms To Go requires a down payment equal to sales tax and delivery.
            </span>
            <span
              className="plantext click-here"
            >
              Click Here for More Details &gt;
            </span>
          </button>
        </div>
        <hr />
        <div
          className="bottom-info grid-x"
        >
          <div
            className="cell medium-8"
          >
            <p>
              Manage Your Existing Rooms To Go Account
            </p>
            <p>
              If you already have a Rooms To Go account and would like to access it,
            </p>
            <p>
              please enter the first six digits of your account number below.
            </p>
            <div
              className="login-list"
            >
              <h3>
                Logging in allows you to:
              </h3>
              <ul>
                <li>
                  Pay your Bill
                </li>
                <li>
                  Check your available balance
                </li>
                <li>
                  Manage your account
                </li>
              </ul>
            </div>
            <div
              className="account-number-lookup cell grid-x"
            >
              <div
                className="cell medium-12"
              >
                <label
                  htmlFor="accountNumber"
                >
                  First Six Digits of Your Account Number
                </label>
              </div>
              <input
                aria-label="accountNumber"
                aria-required="true"
                className="login-field"
                id="accountNumber"
                name="accountNumber"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Account Number"
                type="form"
              />
              <button
                className="account-btn"
                gtm-action="account-login"
                gtm-category="account-input-form"
                id="account-btn"
                name="account-btn"
                onClick={[Function]}
                value="account-number-submit"
              >
                GO
              </button>
            </div>
          </div>
          <div
            className="cell medium-4 cc-img"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="undefinedrtg-credit-cards-l.gif"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CreditOptions renders correctly with an account error 1`] = `
<div
  className="grid-container"
>
  <div
    className="grid-x card options-info"
  >
    <div
      className="small-12 cell grid-x"
    >
      <div
        className="cell medium-4"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="undefinedshopOnline.jpg"
        />
      </div>
      <div
        className="cell medium-8"
      >
        <div
          className="top-info"
        >
          <h2>
            Open A Rooms To Go Account - Credit Extended by Synchrony Bank
          </h2>
          <p>
            With a Rooms To Go account, you will be able to take advantage of exclusive financing offers in your area. Click below to fillout our online application.
          </p>
          <div
            className="cell medium-6 "
          >
            <form
              id="creditApplnForm"
              method="POST"
              name="frmApplicationTermsCond"
              target="_blank"
            >
              <input
                id="uniqueId"
                name="uniqueId"
                type="hidden"
              />
              <input
                id="postbackRefId"
                name="postbackRefId"
                type="hidden"
                value="test-id"
              />
              <input
                name="pcgc"
                type="hidden"
              />
              <input
                name="mid"
                type="hidden"
              />
              <input
                aria-label="Continue with online application (opens in new window)"
                className="blue-action-btn"
                data-seo="applyFinance"
                id="applyFinance"
                role="link"
                type="submit"
                value="Continue with online application"
              />
            </form>
            <p>
              Please note that online finance offers may be different from advertised credit promotions available in showrooms.
            </p>
          </div>
        </div>
        <hr />
        <div
          className="mid-banner"
        >
          <button
            className="link no-margin"
            onClick={[Function]}
          >
            <span
              className="plantitle"
            >
              Interest free financing for 60 months*
            </span>
            <span
              className="plantext"
            >
              On purchases priced at $999.99 and up made with your Rooms To Go credit card through 10/14/19. 
            </span>
            <span
              className="plantext"
            >
              Equal Monthly Payments required for 60 months.* Monthly payments shown are only applicable with this special financing offer.
            </span>
            <span
              className="plantext"
            >
              Rooms To Go requires a down payment equal to sales tax and delivery.
            </span>
            <span
              className="plantext click-here"
            >
              Click Here for More Details &gt;
            </span>
          </button>
        </div>
        <hr />
        <div
          className="bottom-info grid-x"
        >
          <div
            className="cell medium-8"
          >
            <p>
              Manage Your Existing Rooms To Go Account
            </p>
            <p>
              If you already have a Rooms To Go account and would like to access it,
            </p>
            <p>
              please enter the first six digits of your account number below.
            </p>
            <div
              className="login-list"
            >
              <h3>
                Logging in allows you to:
              </h3>
              <ul>
                <li>
                  Pay your Bill
                </li>
                <li>
                  Check your available balance
                </li>
                <li>
                  Manage your account
                </li>
              </ul>
            </div>
            <div
              className="account-number-lookup cell grid-x"
            >
              <div
                className="cell medium-12"
              >
                <label
                  htmlFor="accountNumber"
                >
                  First Six Digits of Your Account Number
                </label>
              </div>
              <p
                className="error"
              >
                test error
              </p>
              <input
                aria-label="accountNumber"
                aria-required="true"
                className="login-field"
                id="accountNumber"
                name="accountNumber"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Account Number"
                type="form"
              />
              <button
                className="account-btn"
                gtm-action="account-login"
                gtm-category="account-input-form"
                id="account-btn"
                name="account-btn"
                onClick={[Function]}
                value="account-number-submit"
              >
                GO
              </button>
            </div>
          </div>
          <div
            className="cell medium-4 cc-img"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="undefinedrtg-credit-cards-l.gif"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CreditOptions renders correctly with default props 1`] = `
<div
  className="grid-container"
>
  <div
    className="grid-x card options-info"
  >
    <div
      className="small-12 cell grid-x"
    >
      <div
        className="cell medium-4"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="undefinedshopOnline.jpg"
        />
      </div>
      <div
        className="cell medium-8"
      >
        <div
          className="top-info"
        >
          <h2>
            Open A Rooms To Go Account - Credit Extended by Synchrony Bank
          </h2>
          <p>
            With a Rooms To Go account, you will be able to take advantage of exclusive financing offers in your area. Click below to fillout our online application.
          </p>
          <div
            className="cell medium-6 "
          >
            <form
              id="creditApplnForm"
              method="POST"
              name="frmApplicationTermsCond"
              target="_blank"
            >
              <input
                id="uniqueId"
                name="uniqueId"
                type="hidden"
              />
              <input
                id="postbackRefId"
                name="postbackRefId"
                type="hidden"
                value="test-id"
              />
              <input
                name="pcgc"
                type="hidden"
              />
              <input
                name="mid"
                type="hidden"
              />
              <input
                aria-label="Continue with online application (opens in new window)"
                className="blue-action-btn"
                data-seo="applyFinance"
                id="applyFinance"
                role="link"
                type="submit"
                value="Continue with online application"
              />
            </form>
            <p>
              Please note that online finance offers may be different from advertised credit promotions available in showrooms.
            </p>
          </div>
        </div>
        <hr />
        <div
          className="mid-banner"
        >
          <button
            className="link no-margin"
            onClick={[Function]}
          >
            <span
              className="plantitle"
            >
              Interest free financing for 60 months*
            </span>
            <span
              className="plantext"
            >
              On purchases priced at $999.99 and up made with your Rooms To Go credit card through 10/14/19. 
            </span>
            <span
              className="plantext"
            >
              Equal Monthly Payments required for 60 months.* Monthly payments shown are only applicable with this special financing offer.
            </span>
            <span
              className="plantext"
            >
              Rooms To Go requires a down payment equal to sales tax and delivery.
            </span>
            <span
              className="plantext click-here"
            >
              Click Here for More Details &gt;
            </span>
          </button>
        </div>
        <hr />
        <div
          className="bottom-info grid-x"
        >
          <div
            className="cell medium-8"
          >
            <p>
              Manage Your Existing Rooms To Go Account
            </p>
            <p>
              If you already have a Rooms To Go account and would like to access it,
            </p>
            <p>
              please enter the first six digits of your account number below.
            </p>
            <div
              className="login-list"
            >
              <h3>
                Logging in allows you to:
              </h3>
              <ul>
                <li>
                  Pay your Bill
                </li>
                <li>
                  Check your available balance
                </li>
                <li>
                  Manage your account
                </li>
              </ul>
            </div>
            <div
              className="account-number-lookup cell grid-x"
            >
              <div
                className="cell medium-12"
              >
                <label
                  htmlFor="accountNumber"
                >
                  First Six Digits of Your Account Number
                </label>
              </div>
              <input
                aria-label="accountNumber"
                aria-required="true"
                className="login-field"
                id="accountNumber"
                name="accountNumber"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Account Number"
                type="form"
              />
              <button
                className="account-btn"
                gtm-action="account-login"
                gtm-category="account-input-form"
                id="account-btn"
                name="account-btn"
                onClick={[Function]}
                value="account-number-submit"
              >
                GO
              </button>
            </div>
          </div>
          <div
            className="cell medium-4 cc-img"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="undefinedrtg-credit-cards-l.gif"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
